hack/e2e-internal/wait-pods-running.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# wait-pods-running waits until all pods in the cluster are 'Running', and exits
+# with an error if this doesn't happen before a deadline.
+
+# Settings:
+# deadline is in seconds, so 1200 = 20 minutes.
+deadline=$(($(date '+%s')+1200))
+# pause is how many seconds to sleep between pod get calls.
+pause=5
+# min_pods is the minimum number of pods we require.
+min_pods=1
+
+# Check pod statuses.
+echo "Waiting for all pods to be 'Running'" >&2
+all_running=0
+until [[ ${all_running} == 1 ]]; do
+  if [[ "$(date '+%s')" -ge "${deadline}" ]]; then
+    echo "All pods never 'Running' in time." >&2
+    exit 1
+  fi
+  statuses=($(${KUBECTL} get pods --template='{{range.items}}{{.currentState.status}} {{end}}'))
```

Comment:
Should we ensure that the system add-on pods are namespaced first and only look for those? (are they?)